staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/metrics/metrics.go

```diff
@@ -31,6 +35,54 @@ const (
 	ToStorageLabel   = "to_storage"
 )
 
+type metricLabels struct {
+	transformationType string
+	providerName       string
+	keyIDHash          string
+}
+type metricLabelsMap struct {
```

Comment:
I do not think we should implement our own cache here, lets just use the LRU cache used by kms v1 for now.

Reply:
Switched to `github.com/golang/groupcache/lru`